learningequality/kolibri
# Disconnection state management

### Summary
 * Adds the Javascript middleware to manage vuex state for disconnection
 * Adds XHR client interceptors to catch error responses that indicate disconnection
 * Adds XHR client interceptor to prevent outbound requests when disconnected
 * Cleans up logout behaviour, and double checks when permission denied responses are received
 * Verified by using the app on various pages and then turning on the 'offline' toggle on the network tab.

### Reviewer guidance
 * To test see above!
 * Check connecting, disconnecting, connecting again.
 * Check the manual 'try to connect' behaviour

### References
 * Finally fixes #1477 

----

### Contributor Checklist

- [x] PR has the correct target milestone
- [x] If PR is ready for review, it has been assigned or requests review from someone and labeled as 'needs review'
- [x] PR does not introduce [accessibility regressions](http://kolibri.readthedocs.io/en/develop/dev/manual_testing.html#accessibility-a11y-testing)

### Reviewer Checklist
